components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts

```diff
@@ -122,33 +150,46 @@ export const useSwap = () => {
   // Mutations
   const [generateSwapQuote] = useGenerateSwapQuoteMutation()
 
-  // Queries
-  const { data: defaultFiatCurrency } = useGetDefaultFiatCurrencyQuery()
-  const { data: supportedNetworks } = useGetSwapSupportedNetworksQuery()
-
   // Memos
   const fromNetwork = useMemo(() => {
-    if (fromToken && supportedNetworks?.length) {
-      return supportedNetworks.find(
-        (network) =>
-          network.chainId === fromToken.chainId &&
-          network.coin === fromToken.coin
-      )
+    if (!supportedNetworks) {
```

Comment:
Use the following since this would not work for an empty array.

```ts
if (!supportedNetworks?.length) {
  ...
}
```

Reply:
Updated!